presenters/thread.rb

```diff
@@ -87,17 +105,28 @@ def get_paged_merged_responses(thread_id, responses, skip, limit, recursive=fals
       content = Comment.where(comment_thread_id: thread_id, "parent_ids" => []).
         where({:id => {"$in" => paged_response_ids}}).sort({"sk" => 1})
     end
-    {"responses" => merge_response_content(content), "response_count" => response_ids.length}
+    {"responses" => merge_response_content(content, thread_id), "response_count" => response_ids.length}
   end
 
   # Takes content output from Mongoid in a depth-first traversal order and
   # returns an array of first-level response hashes with content represented
-  # hierarchically, with a comment's list of children in the key "children".
-  def merge_response_content(content)
+  # hierarchically, with a comment's list of children in the key "children"
+  # and count of children comment in "child_count"
+  def merge_response_content(content, thread_id)
+    parent_child_maps = Comment.where(comment_thread_id: thread_id).map_reduce(@@map, @@reduce).out(inline: true)
```

Comment:
Due to the optimization of not sending in all data, will children.length sometimes be 0 when child_count > 0?  
1. If this is a true statement, then the comments need to reflect this.  For example: "An empty array of children may not reflect that the comment has no children, but just that we didn't load the children."
2. If this statement is not accurate, then it seems like on line 137, right after we add the item_hash to the children array, you could add the following line and drop the new query altogether:
```
ancestry.last["child_count"] = ancestry.last["child_count"] + 1
``` 

If 1. above was correct, and this extra query really is needed, some additional thoughts:
1. I'd rename `parent_child_maps` to `parent_child_count_maps`.
2. I'd see if you could simplify the query by using actual aggregation (e.g. group/sum) rather than map_reduce.  It might perform better, but not sure.  See https://docs.mongodb.org/v3.0/reference/operator/aggregation/sum/#grp._S_sum
3. We will still need performance testing to make a final decision on whether the extra query does perform well enough if it is necessary.